### PR TITLE
USWDS - Password: Replace link with button

### PR DESCRIPTION
**Pull request by amyleadem**
## Summary
**Updated markup in the `usa-password` package to use a `<button>` element instead of an anchor element.** This markup is more semantically appropriate for the related on-page user interaction. 

## Related issue

Closes https://github.com/uswds/uswds/issues/4592

## Preview link

Preview link: [Sign-in pattern template](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-password-button/?path=/story/pages-sign-in--sign-in-page)

## Problem statement
Interactive on-page actions should be controlled by `<button>` elements instead of `<a>` elements. However, our show/hide password functionality in our sign-in template us declared with an `<a>` tag, despite being related to an on-page action. 

## Solution
Replace the `<a>` element with a `<button>` in the sign-in template.

Note: This is not a breaking change. Using the old markup will still render a result that looks and behaves as expected. 

## Testing and review

To review:
1. Open the updated [sign-in template page](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-password-button/?path=/story/pages-sign-in--sign-in-page)
2. Confirm that the button element has the appropriate markup
3. Confirm that the button appearance matches [current style](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/?path=/story/pages-sign-in--sign-in-page)
4. Confirm that the show/hide functionality works as expected by typing anything in the password field and then click on the `Show password` and `Hide password` button. 
5. Confirm that prior markup still looks and behaves as expected by injecting the following via web inspector.
```
<p class="usa-form__note">
  <a title="Toggle password" href="" class="usa-show-password" aria-controls="password-sign-in" data-show-text="Show password" data-hide-text="Hide password">Show password</a>
</p>
```
---

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [x] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

### CODE REVIEW

File: packages/usa-form/src/styles/_usa-form.scss [hunk 1]
```diff
@@ -71,7 +71,8 @@
   }
 }
 
-.usa-form__note {
+.usa-form__note,
+.usa-form__note.usa-button--unstyled {
```
[the remark this answers is not in the document]
Comment (amyleadem) [reply packages/usa-form/src/styles/_usa-form.scss]: Clean. Thanks!
I also included `usa-show-multipassword`, though I am not really sure where/how we use that. 

Updated in [1293e5f](https://github.com/uswds/uswds/pull/4847/commits/1293e5f4a261b12ca1e8eec3a148d70df640d1e3)

[the remark this answers is not in the document]
Comment (mejiaj) [reply packages/usa-form/src/styles/_usa-form.scss]: Great thanks! Where is `usa-show-multipassword` used? If it's a variant for show-password we could do BEM-style `usa-show-password--multi`.

[the remark this answers is not in the document]
Comment (amyleadem) [reply packages/usa-form/src/styles/_usa-form.scss]: I see it only in these files:
https://github.com/uswds/uswds/blob/develop/packages/uswds-core/src/js/utils/test/toggle-form-input.spec.js#L8
https://github.com/uswds/uswds/blob/develop/packages/uswds-core/src/js/utils/test/toggle.template.html#L5
https://github.com/uswds/uswds/blob/develop/packages/_usa-password/src/index.js#L7

It seems to be the same as `usa-show-password`? Am I missing something?

[the remark this answers is not in the document]
Comment (amyleadem) [reply packages/usa-form/src/styles/_usa-form.scss]: @mejiaj I am not seeing that `usa-show-multipassword` is actively in use in our demos or documentation. Looking back through the history, I see that `usa-show-multipassword` is used in the password reset form in [2.13.3](https://github.com/uswds/uswds/blob/release-2.13.3/src/components/templates/password-reset-form.njk#L24), but switches to `usa-show-password` in [3.0.0-beta.1](https://github.com/uswds/uswds/blob/release-3.0.0-beta.1/src/components/_usa-password/usa-password.twig). From that point on `usa-show-multipassword` is only found in tests. 

Do you know anything about the switch? Is multipassword intended to be deprecated? Either way, they appear to be equivalent. Can we leave it as is for now?

[the remark this answers is not in the document]
Comment (mejiaj) [reply packages/usa-form/src/styles/_usa-form.scss]: I think the only difference is in the number of fields. There's a template that has 2 fields, 1 for setting and another for confirming that can be toggled by a single button.

**Review by mejiaj — APPROVED** [review]

**Comment by amyleadem** [comment]
@thisisdano @mejiaj I removed the references to `usa-show-multipassword` in [bd78578](https://github.com/uswds/uswds/pull/4847/commits/bd785788da7e5a58aeb2a3a8a502997e55a5c3b0). Please let me know if you need anything else.

**Pull request merged**